test-data/unit/check-callable.test

```diff
@@ -449,3 +449,11 @@ if callable():  # E: Too few arguments for "callable"
     pass
 
 [builtins fixtures/callable.pyi]
+
+[case testCallableWithNoneArgs]
+
+fn = None
+if callable(fn):
+  fn()
```

Comment:
Can you also add a test case that narrows down from a union of `None` and a callable type, and use `reveal_type(fn)` to display type of the callable in a few different places?